scripts/data/PublishWidgetData.ts

```diff
@@ -0,0 +1,90 @@
+import * as fs from "fs";
+import { join, dirname } from "path";
+import { FileReadError, PatternNotFoundError } from "./errors";
+
+const result = ["packages/pluggableWidgets", "packages/customWidgets"]
+    .reduce<string[]>(
+        (result, packageRoot) => [
+            ...result,
+            ...fs
+                .readdirSync(packageRoot)
+                .filter(path => !path.startsWith("."))
+                .map(dir => join(packageRoot, dir))
+        ],
+        []
+    )
+    .map(packagePath => {
+        try {
+            const { widgetFileName } = extractTextFromFile(packagePath, "src/package.xml", {
+                widgetFileName: [new RegExp('<widgetFile path="(.*\\.xml)"\\s*/>')]
+            });
+
+            const { id, pluginWidget, offlineCapable, supportedPlatform } = extractTextFromFile(
+                packagePath,
+                `src/${widgetFileName}`,
+                {
+                    id: [new RegExp(`id="(.+?)"`)],
+                    pluginWidget: [new RegExp(`pluginWidget="(.+?)"`), false],
+                    offlineCapable: [new RegExp(`offlineCapable="(.+?)"`), false],
+                    supportedPlatform: [new RegExp(`supportedPlatform="(.+?)"`), false]
+                }
```

Comment:
Small thing: I don't think these regexes depend on `packagePath`. Not sure if it has any impact on performance, but could be an idea to move the creations of these regexes to outside of the map so we only create them once.

Reply:
Nice catch. They did rely on it earlier, so it was a leftover.